apps/rush-lib/src/logic/pnpm/PnpmProjectDependencyManifest.ts

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import * as semver from 'semver';
+import {
+  JsonFile,
+  InternalError,
+  FileSystem
+} from '@microsoft/node-core-library';
+
+import {
+  PnpmShrinkwrapFile,
+  IPnpmShrinkwrapDependencyYaml,
+  parsePnpmDependencyKey
+} from './PnpmShrinkwrapFile';
+import { RushConfigurationProject } from '../../api/RushConfigurationProject';
+import { RushConstants } from '../RushConstants';
+import { BasePackage } from '../base/BasePackage';
+import { DependencySpecifier } from '../DependencySpecifier';
+
+export interface IPnpmProjectDependencyManifestOptions {
+  pnpmShrinkwrapFile: PnpmShrinkwrapFile;
+  project: RushConfigurationProject;
+}
+
+/**
+ * This class handles creating the project/.rush/temp/project-dependencies.json file
+ * which tracks the direct and indirect dependencies that a project consumes. This is used
+ * to better determine which projects should be rebuilt when dependencies are updated.
+ */
+export class PnpmProjectDependencyManifest {
+  private static _cache: Map<string, string> = new Map<string, string>();
+
+  private readonly _projectDependencyManifestFilename: string;
+  private readonly _pnpmShrinkwrapFile: PnpmShrinkwrapFile;
+  private readonly _project: RushConfigurationProject;
+
+  private _projectDependencyManifestFile: Map<string, string>;
```

Comment:
Add a comment saying what are the key and value for this cache, and include an example of each.